[new product] add cloud-sql-proxy (#7379)
captn3m0/endoflife.date: products/google-cloud-sql-proxy.md

---

> # Name of the product (mandatory).

Reviewer: ```suggestion
```

title: Cloud SQL Auth Proxy

> category: service
> tags: database google server-app service

Reviewer: ```suggestion
category: app
tags: google
```

Based on what I read cloud-sql-proxy is a tool that may be used on servers, but also for local development. So `app` is a better fit than `server-app`.

cloud-sql-proxy is neither a `service` or a `database`, so removed them.

iconSlug: googlecloud

> permalink: /cloud-sql-auth-proxy

Reviewer: The markdown file must be renamed after the permalink : `cloud-sql-auth-proxy.md`.

alternate_urls:
-   /cloud-sql-proxy
versionCommand: cloud-sql-proxy --version
releasePolicyLink: https://github.com/GoogleCloudPlatform/cloud-sql-proxy?tab=readme-ov-file#support-policy
changelogTemplate: "https://github.com/GoogleCloudPlatform/cloud-sql-proxy/releases/tag/__LATEST__"
eolColumn: Support

> # Auto-update release configuration (optional).
> # This is used for automatically updating `releaseDate`, `latest`, and `latestReleaseDate` for every release.
> # Multiple configurations are allowed.
> # Please visit https://github.com/endoflife-date/endoflife.date/wiki/Automation for more details.
> # The presence of such configuration modifies the product page so that users are informed that existing
> # releases are automatically updated with latest versions.

Reviewer: ```suggestion
```

auto:
  methods:
    - git: https://github.com/GoogleCloudPlatform/cloud-sql-proxy.git

>       regex: ^v(?P<major>\d+)_(?P<minor>\d+)_(?P<patch>\d{1,3})_?(?P<tiny>\d+)?$

Reviewer: ```suggestion
```

Based on https://github.com/GoogleCloudPlatform/cloud-sql-proxy/tags, tags template is `vX.Y.Z`, which is covered by the default regex.

> releases:

Reviewer: Could you document the `1` release as well ?

-   releaseCycle: "2"

>     releaseDate: 2023-01-17

Reviewer 2: this part has to be 2025-04-28 according to here: https://github.com/GoogleCloudPlatform/cloud-sql-proxy/releases/tag/v2.16.0

    eol: 2026-04-28
    latest: "2.16.0"
    latestReleaseDate: 2025-04-28

---

> > [Cloud SQL Auth Proxy](https://cloud.google.com/sql/docs/mysql/sql-proxy) is a service that enables secure, short-lived connections to Cloud SQL databases without exposing them to the public internet.

Reviewer: ```suggestion
> [Cloud SQL Auth Proxy](https://cloud.google.com/sql/docs/mysql/sql-proxy) is a utility for ensuring secure connections to Google Cloud SQL instances.
```

Based on https://github.com/GoogleCloudPlatform/cloud-sql-proxy?tab=readme-ov-file.

> Cloud SQL Auth Proxy supports releases for 1 year after the release date, mentioned in their [support policy](https://github.com/GoogleCloudPlatform/cloud-sql-proxy?tab=readme-ov-file#support-policy).

Reviewer: Could you mention that Cloud SQL Auth Proxy follows [semantic versioning](https://semver.org/) ?

> Cloud SQL Auth Proxy supports releases for 1 year after the release date, mentioned in their [support policy](https://github.com/GoogleCloudPlatform/cloud-sql-proxy?tab=readme-ov-file#support-policy).

Reviewer: Based on https://github.com/GoogleCloudPlatform/cloud-sql-proxy?tab=readme-ov-file#support-policy, the support policy is more complex:

- There is a notion of active release / maintenance release, with different support level between both. This can be modeled in endoflife.date with the eaos / eol fields, see https://endoflife.date/contribute.
- New major versions are guaranteed to be "active" for a minimum of 1 year.
